### ISSUE TITLE
LanguageFeatures/Extension-methods/tearoffs_t03.dart

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/co19#456]
In three places the function `funcFoo`, which is declared to be `int Function(int)`, is invoked with arguments that are not assignable to `int`:
```
  Expect.equals(3, funcFoo([1]));
  Expect.equals(3, funcFoo("[1]"));
  Expect.equals(13, funcFoo("Lily was here"));
```

**Comment by eernstg** [comment]
Drive-by fun fact: When I saw this I thought "I already mentioned that" .. and I did mention the above type errors in #446. But that issue was closed with https://github.com/dart-lang/co19/commit/947e14711f34585181b675fd5addbc02e60b28bb, which turns out to be an update to tearoffs_t02.dart, not 03. ;-)

But the change in tearoffs_t02.dart was still useful, because it eliminated the generic function instantiation that this test used to have in line 23 and 24, and the requirement that torn off functions are not equal is not tested strictly when we also have the generic function instantiation: An implementation could then incorrectly make torn off extension methods equal (e.g., using similar rules as those for instance methods), and the test would still pass because the generic instantiation step still makes them non-equal.

So everything is fine!

**Comment by stereotype441** [comment]
The fix in 896ecaa96762394c200a4ba75dd97865b222673a added a class `X` to `tearoffs1.lib.dart` but didn't update the `export` directive in `tearoffs2.lib.dart`, so now the test fails with the error `The function 'X' isn't defined.`

Also, in `tearoffs1.lib.dart`, the constructor for class `X` is written `C(this.val);` rather than `X(this.val);`, resulting in errors.
